src/test/java/org/jenkinsci/plugins/docker/workflow/DockerDSLTest.java

```diff
@@ -325,6 +325,38 @@ private static void grep(File dir, String text, String prefix, Set<String> match
         });
     }
 
+    @Test public void buildWithMultiStage() {
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                assumeDocker();
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "prj");
+                p.setDefinition(new CpsFlowDefinition(
+                        "node {\n" +
+                                "  sh 'mkdir -p child'\n" +
+                                "  writeFile file: 'child/stuff1', text: 'hello'\n" +
+                                "  writeFile file: 'child/stuff2', text: 'world'\n" +
+                                "  writeFile file: 'child/stuff3', text: env.BUILD_NUMBER\n" +
+                                "  writeFile file: 'child/Dockerfile.other', text: '# This is a test.\\n\\nFROM hello-world AS one\\nARG stuff4=4\\nARG stuff5=5\\nCOPY stuff1 /\\nFROM scratch\\nCOPY --from=one /stuff1 /\\nCOPY stuff2 /\\nCOPY stuff3 /\\n'\n" +
```

Comment:
How about using 
```
"  writeFile file: 'child/Dockerfile.other', text: '''# foo\n" +
  "hey, another line\n" +
  "and another\n" +
  "'''\n" +
```
instead? I don't like looking at really long lines, and within the Pipeline script, you can use the `'''` Groovy multiline string stuff.